### PR TITLE
[fix] future priv

### PR DESCRIPTION
**Pull request by funes79**
## Test Plan

* [ ] acceptance tests

* [x ] Test cases modified `table_grant_test.go` and `view_grant_test.go`

## References

* Fixes https://github.com/chanzuckerberg/terraform-provider-snowflake/issues/279
* Some notes about the fix: as I am not really experienced with the provider I did not find any elegant solution how to exactly test that the `readGenericGrant` is executed on the `ReadTableGrant` or `ReadViewGrant` function call. So I had to reflect the builder interface but that was not enough, as the builder in both cases is FutureGrant. So I made a workaround by using a `validPrivileges` set, in VIEWS it contains only one item, the `SELECT` privilege.

### CODE REVIEW

**Review by ryanking — CHANGES_REQUESTED** [review]
I think this is a reasonable approach. Only think I think should be added is an acceptance test. For that, a configuration that reproduced the problem in #279 would be great.

**Comment by funes79** [comment]
Mhm, how to write and assert acceptance tests?

File: pkg/resources/grant_helpers.go [hunk 1]
```diff
@@ -213,6 +214,18 @@ func readGenericGrant(data *schema.ResourceData, meta interface{}, builder snowf
 	priv := data.Get("privilege").(string)
 	grantOption := data.Get("with_grant_option").(bool)
 
+	// When onlySelect will be true, it means its a future privilege on View, thus we ignore TABLE privileges
```
Comment (ryanking) [inline pkg/resources/grant_helpers.go:217]: What is `onlySelect`?

File: pkg/resources/schema_grant_acceptance_test.go [hunk 2]
```diff
@@ -38,6 +40,15 @@ func TestAccSchemaGrant(t *testing.T) {
...
 			},
+			// TABLE AND VIEW FUTURE GRANTS
+			{
+				Config: futureTableAndViewGrantConfig(sName, roleNameTable, roleNameView),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_view_grant.select_on_future_views", "roles.#", "1"),
+					resource.TestCheckResourceAttr("snowflake_table_grant.select_on_future_tables", "roles.#", "1")
+					resource.TestCheckResourceAttr("snowflake_view_grant.select_on_future_views", "privilege", "SELECT"),
+			//	),
+			//},
```
Comment (ryanking) [inline pkg/resources/schema_grant_acceptance_test.go:51]: This does not appear to be syntactically correct.

**Review by ryanking — CHANGES_REQUESTED** [review]

File: pkg/resources/grant_helpers.go (hunk 1 above)
Comment (funes79) [reply pkg/resources/grant_helpers.go]: Old comment, pushed a newer version.

File: pkg/resources/schema_grant_acceptance_test.go (hunk 2 above)
Comment (funes79) [reply pkg/resources/schema_grant_acceptance_test.go]: Yes, fixed in newer commit.

**Comment by funes79** [comment]
Just a side note about acceptance tests: All tests should be written with UPPERCASE ids as in my test because some clients can have `QUOTED_IDENTIFIERS_IGNORE_CASE` on Account level, so if the acceptance test creates a database `test_abc123` then the assertion fails that `TEST_ABC123` is not equal to `test_abc123`

**Comment by funes79** [comment]
@ryanking please review again. Thank you

**Pull request merged**